packages/flutter_tools/lib/src/project.dart

```diff
@@ -486,9 +487,15 @@ class AndroidProject extends FlutterProjectPlatform {
   }
 
   File get appManifestFile {
-    return isUsingGradle
```

Comment:
@christopherfujino I had mismatching results when originally running this test.
The `fileSystem` created in the test would report that the `appManifestFile` exists, but the `globals.fs` file system in this getter reported that it did not exist.

According to a doc comment I can override `globals.fs` in tests. I don't see the `setUp((){})` doing something like that. What would be the right thing to do here?

A) Override `globals.fs` in the test
B) Fix the getter to use `hostAppGradleRoot`, which does use the `fileSystem` that was set up in the specific test (which is what I did here)

Reply:
Good find. Looks like whatever test you copy-pasted this from was also leaking the real file system. The whole purpose of the `testUsingContext()` function you're calling is that it allows you to provide `overrides` to the Zone/context getters (`globals.fs` is essentially calling `context.get<FileSystem>();).

Here is an example of constructing a FileSystem in `setUp` and then providing it as a context override for a `testUsingContext` invocation: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/hermetic/build_test.dart#L60

Reply:
although if it makes sense, what you did here is probably better (it looks cleaner)

Reply:
Thanks for the tip, I'll update the test with an override for the file system.